src/loaders/utils.js

```diff
@@ -102,3 +104,75 @@ export function isBioformatsNoPadHeightVersion(software) {
   }
   return false;
 }
+
+/**
+ * Returns actual image stats for static imagery and an estimate via a downsampled version of image pyramids.
+ * This is helpful for generating histograms of your channel data, or scaling your sliders down to a reasonable range.
+ * @param {Object} args
+ * @param {Object} args.loader A valid loader object.
+ * @param {Array} args.loaderSelection Array of valid dimension selections
+ * @returns {Object} { means, dataRanges, standardDeviations, data, firstQuartiles, thirdQuartiles }, each arrays with entries for each channel based on your selection.
+ */
+export async function getChannelStats({ loader, loaderSelection }) {
+  const z = loader.isPyramid ? loader.numLevels - 1 : 0;
+  const rasters = await loader.getRaster({ z, loaderSelection });
+  const { data } = rasters;
+  const channelStats = {
+    means: [],
+    medians: [],
+    dataRanges: [],
+    standardDeviations: [],
+    data,
+    firstQuartiles: [],
+    thirdQuartiles: []
```

Comment:
Why include `data` in stats? I think having an array of `stats` objects would be a better data structure for this (compared to an object of arrays). The UI components use channel objects in Vitessce and then serialize to arrays only for rendering purposes with viv.

You have 4 channels, return the stats as objects for those channels. If an end user wanted stats for channel `i` accessing those would look like:

```javascript
const stats = {
  mean: channelStats.means[i],
  median: channelStats.medians[i],
  dataRange: channelStats.dataRanges[i],
  standardDeviation: channelStats.standardDeviations[i],
  firstQuartile: channelStats.firstQuartiles[i],
  thirdQuartile: channelStats.thirdQuartiles[i],
}
```

vs:

```javascript
const stats = channelStats[i];
const { mean, median, dataRange, standardDeviation, firstQuartile, thirdQuartile } = stats;
```

The former is much more error prone.

Reply:
nit, but some more convenient / common keys might be useful:

- `dataRange` --> `domain` (https://github.com/d3/d3-scale#continuous_domain)
- `standardDeviation` --> `sd`
- `firstQuartile` --> `q1`
- `thirdQuartile` --> `q3`

Reply:
You're right about the data structure.  Will do.

Reply:
As for `data`, it will be helpful for generating visualizations of the data along with stats, like a histogram.